From e9a76b1fcb751e8a907a871df5cdf60f3a0a1b65 Mon Sep 17 00:00:00 2001
From: Daniel Emminizer <daniel.m.emminizer.civ@us.navy.mil>
Date: Wed, 30 Apr 2025 09:18:56 +0000
Subject: [PATCH 28/29] SDK: Remove GLEW dependency

**Release Notes:** SDK NEW FEATURE: GLEW third party dependency removed with a Dear ImGui version update.

**JIRA Issue:** SIM-18212

**Description:** Removed GLEW references, now using ImGui default GL initialization.

**Testing Performed:** Build and run several examples on both Windows and Linux with latest osgEarth master, and with current osgEarth drop.
---
 Examples/ASIViewer/ASIViewer.cpp              |  2 --
 Examples/AngleTest/AngleTestExample.cpp       |  2 --
 .../AntennaPattern/ExampleAntennaPattern.cpp  |  2 --
 .../ExampleArticulatedModel.cpp               |  2 --
 Examples/AsyncModelLoading/main.cpp           |  2 --
 Examples/BasicViewer/BasicViewer.cpp          |  2 --
 Examples/BeamTest/BeamTest.cpp                |  2 --
 Examples/CentroidEyePosition/main.cpp         |  2 --
 .../EciTrackHistoryTest.cpp                   |  2 --
 Examples/FragmentEffectTest/main.cpp          |  2 --
 .../GOGAttachments/ExampleGOGAttachments.cpp  |  2 --
 Examples/GOGReader/GOGReader.cpp              |  2 --
 Examples/GateTest/GateTest.cpp                |  2 --
 Examples/GeoFencing/ExampleGeoFencing.cpp     |  2 --
 .../HudPositionManager/HudPositionTest.cpp    |  2 --
 Examples/ImageIcons/ImageIcons.cpp            |  2 --
 Examples/LoadEarthFile/LoadEarthFile.cpp      |  2 --
 Examples/LocalGrid/LocalGrid.cpp              |  2 --
 Examples/LocatorTest/LocatorTestExample.cpp   |  2 --
 Examples/MapScale/MapScale.cpp                |  2 --
 Examples/MassiveData/MassiveData.cpp          |  2 --
 Examples/Ocean/Ocean.cpp                      |  2 --
 Examples/Overhead/OverheadExample.cpp         |  2 --
 Examples/Picking/ExamplePicking.cpp           |  2 --
 .../PlanetariumViewTest.cpp                   |  2 --
 .../PlatformAzimElevViewTest.cpp              |  2 --
 .../PlatformSymbology/PlatformSymbology.cpp   |  2 --
 Examples/Projectors/Projectors.cpp            |  2 --
 Examples/RCS/ExampleRCS.cpp                   |  2 --
 Examples/RFProp/RFProp.cpp                    |  2 --
 Examples/RadialLOS/ExampleRadialLOS.cpp       |  2 --
 Examples/RangeTool/ExampleRangeTool.cpp       |  2 --
 Examples/SimpleServer/ViewerApp.cpp           |  2 --
 Examples/SkyModel/ExampleSkyModel.cpp         |  2 --
 .../ExampleTimestampedLayer.cpp               |  2 --
 .../TrackHistoryTest/TrackHistoryTest.cpp     |  2 --
 Examples/VelocityLayerTest/main.cpp           |  2 --
 Examples/imgui/CMakeLists.txt                 | 10 ++-----
 Examples/imgui/OsgImGuiHandler.cpp            | 20 -------------
 Examples/imgui/OsgImGuiHandler.h              | 30 -------------------
 INSTALL.md                                    |  1 -
 SDK/simVis/simVisConfig.cmake.in              |  5 ----
 42 files changed, 2 insertions(+), 138 deletions(-)

diff --git a/Examples/ASIViewer/ASIViewer.cpp b/Examples/ASIViewer/ASIViewer.cpp
index be2d220d..90efe7cd 100644
--- a/Examples/ASIViewer/ASIViewer.cpp
+++ b/Examples/ASIViewer/ASIViewer.cpp
@@ -1048,8 +1048,6 @@ int main(int argc, char **argv)
 
   /// show the instructions overlay
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(app));
diff --git a/Examples/AngleTest/AngleTestExample.cpp b/Examples/AngleTest/AngleTestExample.cpp
index aaa4951d..435e43ff 100644
--- a/Examples/AngleTest/AngleTestExample.cpp
+++ b/Examples/AngleTest/AngleTestExample.cpp
@@ -323,8 +323,6 @@ int main(int argc, char **argv)
   viewer->getMainView()->setFocalOffsets(0, -45, 4e5);
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new ::GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   ::GUI::OsgImGuiHandler* gui = new ::GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(dataStore, s_id));
diff --git a/Examples/AntennaPattern/ExampleAntennaPattern.cpp b/Examples/AntennaPattern/ExampleAntennaPattern.cpp
index d2e118c4..e827fa64 100644
--- a/Examples/AntennaPattern/ExampleAntennaPattern.cpp
+++ b/Examples/AntennaPattern/ExampleAntennaPattern.cpp
@@ -579,8 +579,6 @@ int main(int argc, char **argv)
     scene->getSkyNode()->setDateTime(osgEarth::DateTime(2012, 0, 0, 11.0));
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new ::GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   ::GUI::OsgImGuiHandler* gui = new ::GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
 
diff --git a/Examples/ArticulatedModel/ExampleArticulatedModel.cpp b/Examples/ArticulatedModel/ExampleArticulatedModel.cpp
index ee5220bb..b5dcf20c 100644
--- a/Examples/ArticulatedModel/ExampleArticulatedModel.cpp
+++ b/Examples/ArticulatedModel/ExampleArticulatedModel.cpp
@@ -446,8 +446,6 @@ int main(int argc, char **argv)
   viewer->getMainView()->setFocalOffsets(135, -8, 30);
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new ::GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   ::GUI::OsgImGuiHandler* gui = new ::GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(node2.get()));
diff --git a/Examples/AsyncModelLoading/main.cpp b/Examples/AsyncModelLoading/main.cpp
index 1b43ec73..e050033e 100644
--- a/Examples/AsyncModelLoading/main.cpp
+++ b/Examples/AsyncModelLoading/main.cpp
@@ -660,8 +660,6 @@ int main(int argc, char **argv)
   viewer->addEventHandler(new MenuHandler(app));
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(app));
diff --git a/Examples/BasicViewer/BasicViewer.cpp b/Examples/BasicViewer/BasicViewer.cpp
index 3598a039..9985051f 100644
--- a/Examples/BasicViewer/BasicViewer.cpp
+++ b/Examples/BasicViewer/BasicViewer.cpp
@@ -358,8 +358,6 @@ int main(int argc, char** argv)
   hud->addOverlayControl(createHelp());
 
 #else
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   mainView->getEventHandlers().push_front(gui);
 
diff --git a/Examples/BeamTest/BeamTest.cpp b/Examples/BeamTest/BeamTest.cpp
index fd98525b..f3816d13 100644
--- a/Examples/BeamTest/BeamTest.cpp
+++ b/Examples/BeamTest/BeamTest.cpp
@@ -717,8 +717,6 @@ int main(int argc, char** argv)
   viewer->getMainView()->setFocalOffsets(-45, -45, 500.0);
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(dataStore, beamId, viewer->getMainView()));
diff --git a/Examples/CentroidEyePosition/main.cpp b/Examples/CentroidEyePosition/main.cpp
index 9138b0c0..e091ddd7 100644
--- a/Examples/CentroidEyePosition/main.cpp
+++ b/Examples/CentroidEyePosition/main.cpp
@@ -487,8 +487,6 @@ int main(int argc, char **argv)
   app.mainView->setFocalOffsets(270, -20, 650);
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   app.mainView->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(app));
diff --git a/Examples/EciTrackHistoryTest/EciTrackHistoryTest.cpp b/Examples/EciTrackHistoryTest/EciTrackHistoryTest.cpp
index bedfe3bd..82bc1503 100644
--- a/Examples/EciTrackHistoryTest/EciTrackHistoryTest.cpp
+++ b/Examples/EciTrackHistoryTest/EciTrackHistoryTest.cpp
@@ -515,8 +515,6 @@ int main(int argc, char **argv)
   viewer->getMainView()->tetherCamera(platformModel.get());
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(dataStore, platformId, simHandler.get(), viewer->getMainView(), platformModel.get()));
diff --git a/Examples/FragmentEffectTest/main.cpp b/Examples/FragmentEffectTest/main.cpp
index 51089e2d..18aaad5b 100644
--- a/Examples/FragmentEffectTest/main.cpp
+++ b/Examples/FragmentEffectTest/main.cpp
@@ -414,8 +414,6 @@ int main(int argc, char **argv)
   mainView->addEventHandler(new PrefsKeyEventHandler('/', dataStore, platform, scaleDown));
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new ::GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   ::GUI::OsgImGuiHandler* gui = new ::GUI::OsgImGuiHandler();
   mainView->getEventHandlers().push_front(gui);
   auto* panel = new ControlPanel(dataStore, platform);
diff --git a/Examples/GOGAttachments/ExampleGOGAttachments.cpp b/Examples/GOGAttachments/ExampleGOGAttachments.cpp
index 958db9d0..cea0aa51 100644
--- a/Examples/GOGAttachments/ExampleGOGAttachments.cpp
+++ b/Examples/GOGAttachments/ExampleGOGAttachments.cpp
@@ -578,8 +578,6 @@ int main(int argc, char **argv)
 #endif
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new ::GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   ::GUI::OsgImGuiHandler* gui = new ::GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel());
diff --git a/Examples/GOGReader/GOGReader.cpp b/Examples/GOGReader/GOGReader.cpp
index cf331a15..0eb621d1 100644
--- a/Examples/GOGReader/GOGReader.cpp
+++ b/Examples/GOGReader/GOGReader.cpp
@@ -731,8 +731,6 @@ int main(int argc, char** argv)
       attach ? platform.get() : nullptr);
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new ::GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   ::GUI::OsgImGuiHandler* gui = new ::GUI::OsgImGuiHandler();
   mainView->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(*mouseHandler.get()));
diff --git a/Examples/GateTest/GateTest.cpp b/Examples/GateTest/GateTest.cpp
index 901a784e..329cdd6e 100644
--- a/Examples/GateTest/GateTest.cpp
+++ b/Examples/GateTest/GateTest.cpp
@@ -741,8 +741,6 @@ int main(int argc, char** argv)
   viewer->getMainView()->tetherCamera(platformModel.get());
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(dataStore, platformId, gateId, viewer->getMainView(), scene->getScenario()));
diff --git a/Examples/GeoFencing/ExampleGeoFencing.cpp b/Examples/GeoFencing/ExampleGeoFencing.cpp
index 5e2930dc..36f30f74 100644
--- a/Examples/GeoFencing/ExampleGeoFencing.cpp
+++ b/Examples/GeoFencing/ExampleGeoFencing.cpp
@@ -425,8 +425,6 @@ int main(int argc, char **argv)
   buildFences(app, viewer->getSceneManager());
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(app));
diff --git a/Examples/HudPositionManager/HudPositionTest.cpp b/Examples/HudPositionManager/HudPositionTest.cpp
index 3e9278da..5ef92a2e 100644
--- a/Examples/HudPositionManager/HudPositionTest.cpp
+++ b/Examples/HudPositionManager/HudPositionTest.cpp
@@ -506,8 +506,6 @@ int main(int argc, char** argv)
   }
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   mainView->getEventHandlers().push_front(gui);
   ControlPanel* controlPanel = new ControlPanel(hudEditor, dataStore);
diff --git a/Examples/ImageIcons/ImageIcons.cpp b/Examples/ImageIcons/ImageIcons.cpp
index a91916a0..0fe880bc 100644
--- a/Examples/ImageIcons/ImageIcons.cpp
+++ b/Examples/ImageIcons/ImageIcons.cpp
@@ -422,8 +422,6 @@ int main(int argc, char **argv)
   viewer->addEventHandler(new simVis::PopupHandler(scene.get()));
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(dataStore, obj1));
diff --git a/Examples/LoadEarthFile/LoadEarthFile.cpp b/Examples/LoadEarthFile/LoadEarthFile.cpp
index 6c4a049b..28930961 100644
--- a/Examples/LoadEarthFile/LoadEarthFile.cpp
+++ b/Examples/LoadEarthFile/LoadEarthFile.cpp
@@ -434,8 +434,6 @@ int main(int argc, char** argv)
   mouseDispatcher.reset(new simUtil::MouseDispatcher);
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(viewer.get(), createInsetsHandler.get(), mouseDispatcher.get()));
diff --git a/Examples/LocalGrid/LocalGrid.cpp b/Examples/LocalGrid/LocalGrid.cpp
index ca0f09e6..076ff3a3 100644
--- a/Examples/LocalGrid/LocalGrid.cpp
+++ b/Examples/LocalGrid/LocalGrid.cpp
@@ -395,8 +395,6 @@ int main(int argc, char** argv)
   simData::ObjectId platformId = createPlatform(dataStore);
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(dataStore, platformId));
diff --git a/Examples/LocatorTest/LocatorTestExample.cpp b/Examples/LocatorTest/LocatorTestExample.cpp
index f6c29df7..4ac2b320 100644
--- a/Examples/LocatorTest/LocatorTestExample.cpp
+++ b/Examples/LocatorTest/LocatorTestExample.cpp
@@ -641,8 +641,6 @@ int main(int argc, char **argv)
   const auto* mapSRS = viewer->getSceneManager()->getMap()->getSRS();
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new ::GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   ::GUI::OsgImGuiHandler* gui = new ::GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(mapSRS, graph.get()));
diff --git a/Examples/MapScale/MapScale.cpp b/Examples/MapScale/MapScale.cpp
index 319d5831..5135348e 100644
--- a/Examples/MapScale/MapScale.cpp
+++ b/Examples/MapScale/MapScale.cpp
@@ -564,8 +564,6 @@ int main(int argc, char** argv)
   mapScale->bindToFocusManager(mainView->getFocusManager());
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   mainView->getEventHandlers().push_front(gui);
 
diff --git a/Examples/MassiveData/MassiveData.cpp b/Examples/MassiveData/MassiveData.cpp
index bed375f4..b5140361 100644
--- a/Examples/MassiveData/MassiveData.cpp
+++ b/Examples/MassiveData/MassiveData.cpp
@@ -384,8 +384,6 @@ int main(int argc, char** argv)
   viewer->addEventHandler(popupHandler);
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(app, duration));
diff --git a/Examples/Ocean/Ocean.cpp b/Examples/Ocean/Ocean.cpp
index 5c933276..73da8a95 100644
--- a/Examples/Ocean/Ocean.cpp
+++ b/Examples/Ocean/Ocean.cpp
@@ -1322,8 +1322,6 @@ int main(int argc, char** argv)
   // install the handler for the demo keys in the notify() above
   viewer->addEventHandler(new MenuHandler(viewer.get(), scene.get(), menu));
 #else
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new ::GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   ::GUI::OsgImGuiHandler* gui = new ::GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(simpleOceanLayer.get(), tritonLayer.get(), buoyancyCallback.get(),
diff --git a/Examples/Overhead/OverheadExample.cpp b/Examples/Overhead/OverheadExample.cpp
index 2cff44b6..e597859d 100644
--- a/Examples/Overhead/OverheadExample.cpp
+++ b/Examples/Overhead/OverheadExample.cpp
@@ -717,8 +717,6 @@ int main(int argc, char** argv)
   mouseManip->addListener(latLonElevListener.get(), showElevation);
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   // Adjusted projection matrix is incorrect in ortho mode
   gui->setAutoAdjustProjectionMatrix(false);
diff --git a/Examples/Picking/ExamplePicking.cpp b/Examples/Picking/ExamplePicking.cpp
index 71fc8e97..bdf2df63 100644
--- a/Examples/Picking/ExamplePicking.cpp
+++ b/Examples/Picking/ExamplePicking.cpp
@@ -752,8 +752,6 @@ int main(int argc, char** argv)
   }
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewMan->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewMan->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
 
   // Because RTT requires rendering the view to a texture, ImGui would get called twice (and assert)
diff --git a/Examples/PlanetariumViewTest/PlanetariumViewTest.cpp b/Examples/PlanetariumViewTest/PlanetariumViewTest.cpp
index 8fd52448..eea66944 100644
--- a/Examples/PlanetariumViewTest/PlanetariumViewTest.cpp
+++ b/Examples/PlanetariumViewTest/PlanetariumViewTest.cpp
@@ -787,8 +787,6 @@ int main(int argc, char** argv)
   // set up the controls
   osg::observer_ptr<simVis::View> view = viewer->getMainView();
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(app));
diff --git a/Examples/PlatformAzimElevViewTest/PlatformAzimElevViewTest.cpp b/Examples/PlatformAzimElevViewTest/PlatformAzimElevViewTest.cpp
index 8a96073e..e3cc4b5d 100644
--- a/Examples/PlatformAzimElevViewTest/PlatformAzimElevViewTest.cpp
+++ b/Examples/PlatformAzimElevViewTest/PlatformAzimElevViewTest.cpp
@@ -431,8 +431,6 @@ int main(int argc, char **argv)
   // set up the controls
   osg::observer_ptr<simVis::View> view = viewer->getMainView();
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(app));
diff --git a/Examples/PlatformSymbology/PlatformSymbology.cpp b/Examples/PlatformSymbology/PlatformSymbology.cpp
index 8ebebba1..60e1c653 100644
--- a/Examples/PlatformSymbology/PlatformSymbology.cpp
+++ b/Examples/PlatformSymbology/PlatformSymbology.cpp
@@ -1085,8 +1085,6 @@ int main(int argc, char **argv)
 #ifdef HAVE_IMGUI
   ControlPanel* controlPanel = new ControlPanel();
   menuHandler->setImGuiControlPanel(controlPanel);
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new ::GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   ::GUI::OsgImGuiHandler* gui = new ::GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(controlPanel);
diff --git a/Examples/Projectors/Projectors.cpp b/Examples/Projectors/Projectors.cpp
index 610d6412..3581556c 100644
--- a/Examples/Projectors/Projectors.cpp
+++ b/Examples/Projectors/Projectors.cpp
@@ -605,8 +605,6 @@ int main(int argc, char **argv)
   viewer->addEventHandler(new simVis::PopupHandler(scene.get()));
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel());
diff --git a/Examples/RCS/ExampleRCS.cpp b/Examples/RCS/ExampleRCS.cpp
index 83b218e9..a0017ae2 100644
--- a/Examples/RCS/ExampleRCS.cpp
+++ b/Examples/RCS/ExampleRCS.cpp
@@ -329,8 +329,6 @@ int main(int argc, char **argv)
   AppData app;
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new ::GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   ::GUI::OsgImGuiHandler* gui = new ::GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(app));
diff --git a/Examples/RFProp/RFProp.cpp b/Examples/RFProp/RFProp.cpp
index bc34b84c..7e0e493e 100644
--- a/Examples/RFProp/RFProp.cpp
+++ b/Examples/RFProp/RFProp.cpp
@@ -1010,8 +1010,6 @@ int main(int argc, char** argv)
   controlPanel->heatColorProvider = new simRF::GradientColorProvider();
   controlPanel->heatColorProvider->setColorMap(heatColors);
 
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new ::GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   ::GUI::OsgImGuiHandler* gui = new ::GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   controlPanel->minHeight = minHeight;
diff --git a/Examples/RadialLOS/ExampleRadialLOS.cpp b/Examples/RadialLOS/ExampleRadialLOS.cpp
index bacf4619..be0205ce 100644
--- a/Examples/RadialLOS/ExampleRadialLOS.cpp
+++ b/Examples/RadialLOS/ExampleRadialLOS.cpp
@@ -478,8 +478,6 @@ int main(int argc, char **argv)
     osgEarth::Viewpoint("Start", RLOS_LON, RLOS_LAT, RLOS_ALT, 0.0, -45.0, INIT_RANGE_MAX*2000.0));
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new ::GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   ::GUI::OsgImGuiHandler* gui = new ::GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(app));
diff --git a/Examples/RangeTool/ExampleRangeTool.cpp b/Examples/RangeTool/ExampleRangeTool.cpp
index d7bdc44d..45f59081 100644
--- a/Examples/RangeTool/ExampleRangeTool.cpp
+++ b/Examples/RangeTool/ExampleRangeTool.cpp
@@ -668,8 +668,6 @@ int main(int argc, char **argv)
 #ifdef HAVE_IMGUI
   ControlPanel* controlPanel = new ControlPanel();
   menuHandler->setControlPanel(controlPanel);
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(controlPanel);
diff --git a/Examples/SimpleServer/ViewerApp.cpp b/Examples/SimpleServer/ViewerApp.cpp
index 8130ccf7..8f6c675a 100644
--- a/Examples/SimpleServer/ViewerApp.cpp
+++ b/Examples/SimpleServer/ViewerApp.cpp
@@ -367,8 +367,6 @@ void ViewerApp::init_(osg::ArgumentParser& args)
   loadGog_(EXAMPLE_GOG_MISSILE_LLA);
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewManager_->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewManager_->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   mainView->getEventHandlers().push_front(gui);
 
diff --git a/Examples/SkyModel/ExampleSkyModel.cpp b/Examples/SkyModel/ExampleSkyModel.cpp
index 736c8131..612a2dc9 100644
--- a/Examples/SkyModel/ExampleSkyModel.cpp
+++ b/Examples/SkyModel/ExampleSkyModel.cpp
@@ -421,8 +421,6 @@ int main(int argc, char** argv)
   arguments.read("--slpath", slpathArg);
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewMan->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewMan->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   mainView->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(app));
diff --git a/Examples/TimestampedLayer/ExampleTimestampedLayer.cpp b/Examples/TimestampedLayer/ExampleTimestampedLayer.cpp
index 6826e116..573bb180 100644
--- a/Examples/TimestampedLayer/ExampleTimestampedLayer.cpp
+++ b/Examples/TimestampedLayer/ExampleTimestampedLayer.cpp
@@ -306,8 +306,6 @@ int main(int argc, char** argv)
   }
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(app));
diff --git a/Examples/TrackHistoryTest/TrackHistoryTest.cpp b/Examples/TrackHistoryTest/TrackHistoryTest.cpp
index 86628f2f..b25f0cfa 100644
--- a/Examples/TrackHistoryTest/TrackHistoryTest.cpp
+++ b/Examples/TrackHistoryTest/TrackHistoryTest.cpp
@@ -695,8 +695,6 @@ int main(int argc, char **argv)
 
   /// show the instructions overlay
 #ifdef HAVE_IMGUI
-    // Pass in existing realize operation as parent op, parent op will be called first
-  viewer->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewer->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   viewer->getMainView()->getEventHandlers().push_front(gui);
   gui->add(new ControlPanel(app));
diff --git a/Examples/VelocityLayerTest/main.cpp b/Examples/VelocityLayerTest/main.cpp
index d2b03dca..6e5f932f 100644
--- a/Examples/VelocityLayerTest/main.cpp
+++ b/Examples/VelocityLayerTest/main.cpp
@@ -445,8 +445,6 @@ int main(int argc, char* argv[])
   mainView->addEventHandler(stats);
 
 #ifdef HAVE_IMGUI
-  // Pass in existing realize operation as parent op, parent op will be called first
-  viewMan->getViewer()->setRealizeOperation(new GUI::OsgImGuiHandler::RealizeOperation(viewMan->getViewer()->getRealizeOperation()));
   GUI::OsgImGuiHandler* gui = new GUI::OsgImGuiHandler();
   mainView->getEventHandlers().push_front(gui);
 
diff --git a/Examples/imgui/CMakeLists.txt b/Examples/imgui/CMakeLists.txt
index 431b7bde..b35c9a5b 100644
--- a/Examples/imgui/CMakeLists.txt
+++ b/Examples/imgui/CMakeLists.txt
@@ -1,9 +1,4 @@
-if(NOT TARGET simVis)
-    return()
-endif()
-
-find_package(GLEW)
-if(NOT TARGET GLEW::GLEW OR NOT TARGET osgEarth::osgEarth OR NOT TARGET OpenGL::GL)
+if(NOT TARGET simVis OR NOT TARGET osgEarth::osgEarth OR NOT TARGET OpenGL::GL)
     return()
 endif()
 
@@ -41,7 +36,7 @@ add_library(imgui_gl3 STATIC
     ${ImGui_DIR}/backends/imgui_impl_opengl3_loader.h
 )
 target_include_directories(imgui_gl3 PUBLIC "${ImGui_DIR}")
-target_link_libraries(imgui_gl3 PUBLIC GLEW::GLEW OpenGL::GL PRIVATE ${CMAKE_DL_LIBS})
+target_link_libraries(imgui_gl3 PUBLIC OpenGL::GL PRIVATE ${CMAKE_DL_LIBS})
 add_library(VSI::imgui_gl3 ALIAS imgui_gl3)
 
 # Set up the ImGuiLib, based on ImGui
@@ -57,7 +52,6 @@ set(SOURCE_FILES
 add_library(ImGuiLib ${SOURCE_FILES})
 target_include_directories(ImGuiLib PUBLIC .)
 target_link_libraries(ImGuiLib PUBLIC osgEarth::osgEarth VSI::imgui_gl3 PRIVATE simVis simCore simNotify)
-target_compile_definitions(ImGuiLib PRIVATE IMGUI_IMPL_OPENGL_LOADER_GLEW)
 if(CMAKE_COMPILER_IS_GNUCXX)
     target_compile_options(ImGuiLib PRIVATE -fno-strict-aliasing)
 endif()
diff --git a/Examples/imgui/OsgImGuiHandler.cpp b/Examples/imgui/OsgImGuiHandler.cpp
index e1bf501d..d6bbe413 100644
--- a/Examples/imgui/OsgImGuiHandler.cpp
+++ b/Examples/imgui/OsgImGuiHandler.cpp
@@ -20,7 +20,6 @@
  * disclose, or release this software.
  *
  */
-#include "GL/glew.h"
 #include <osg/Camera>
 #include <osg/RenderInfo>
 #include "imgui.h"
@@ -88,25 +87,6 @@
 
 namespace GUI {
 
-GlewInitOperation::GlewInitOperation()
-  : osg::Operation("GlewInitCallback", false)
-{
-}
-
-void GlewInitOperation::operator()(osg::Object* object)
-{
-  osg::GraphicsContext* context = dynamic_cast<osg::GraphicsContext*>(object);
-  if (!context)
-    return;
-
-  if (glewInit() != GLEW_OK)
-  {
-    SIM_ERROR << "glewInit() failed" << std::endl;
-  }
-}
-
-///////////////////////////////////////////////////////////////////////////////
-
 struct OsgImGuiHandler::ImGuiNewFrameCallback : public osg::Camera::DrawCallback
 {
   explicit ImGuiNewFrameCallback(OsgImGuiHandler& handler)
diff --git a/Examples/imgui/OsgImGuiHandler.h b/Examples/imgui/OsgImGuiHandler.h
index e53794f8..2a67b1dd 100644
--- a/Examples/imgui/OsgImGuiHandler.h
+++ b/Examples/imgui/OsgImGuiHandler.h
@@ -39,15 +39,6 @@ namespace GUI {
 
 class BaseGui;
 
-class GlewInitOperation : public osg::Operation
-{
-public:
-  GlewInitOperation();
-  void operator()(osg::Object* object) override;
-};
-
-/////////////////////////////////////////////////////////////////////////
-
 class OsgImGuiHandler : public osgGA::GUIEventHandler
 {
 public:
@@ -65,27 +56,6 @@ public:
   /** Add deprecated GUI to the manager. TODO: Remove once ::GUI::BaseGUI is removed */
   void add(::GUI::BaseGui* gui);
 
-  class RealizeOperation : public ::GUI::GlewInitOperation
-  {
-  public:
-    /** Constructor. If passed a valid operation for parentOp, its operator() will be called first */
-    explicit RealizeOperation(osg::Operation* parentOp = nullptr)
-      : ::GUI::GlewInitOperation(),
-      parentOp_(parentOp)
-    {
-    }
-
-  private:
-    void operator()(osg::Object* object) override
-    {
-      if (parentOp_.valid())
-        parentOp_->operator()(object);
-      GlewInitOperation::operator()(object);
-    }
-
-    osg::ref_ptr<osg::Operation> parentOp_;
-  };
-
   /** Get a pointer to the default font, may be NULL */
   ImFont* getDefaultFont() const;
   /** Get a pointer to the large font, may be NULL */
diff --git a/INSTALL.md b/INSTALL.md
index aafecd75..a0b50a44 100644
--- a/INSTALL.md
+++ b/INSTALL.md
@@ -37,7 +37,6 @@ Third Party Dependencies
 The SIMDIS SDK depends on the following third party libraries:
 
 - [EnTT 3.14+](https://github.com/skypjack/entt)
-- [GLEW 2.1+](https://glew.sourceforge.net/)
 - [OpenSceneGraph 3.6+](https://www.openscenegraph.org)
 - [osgEarth 3.5+](https://www.osgearth.org)
 - [protobuf 2.6+](https://code.google.com/p/protobuf)
diff --git a/SDK/simVis/simVisConfig.cmake.in b/SDK/simVis/simVisConfig.cmake.in
index c415b0b2..7749aec8 100644
--- a/SDK/simVis/simVisConfig.cmake.in
+++ b/SDK/simVis/simVisConfig.cmake.in
@@ -1,10 +1,5 @@
 include(CMakeFindDependencyMacro)
 
-# GLEW needs variable set
-if(NOT "x@GLEW_USE_STATIC_LIBS@" STREQUAL "x")
-    set(GLEW_USE_STATIC_LIBS ON)
-endif()
-
 find_dependency(simNotify)
 find_dependency(simCore)
 find_dependency(simData)
-- 
2.47.1.windows.2

